File: sentimental-credit/credit.py
from cs50 import get_int


def main():
    card = get_int("Number: ")

    sum = 0
    lenght = 0

    # Multiply every other digit by 2 then sums the results
    temp_number = card // 10
    while temp_number > 0:
        product = temp_number % 10 * 2
        while product > 0:
            sum += product % 10
            product //= 10
        temp_number //= 100
        lenght += 1

    # Add the previous sum to the sum of the digits that weren’t multiplied by 2
    temp_number = card
    while temp_number > 0:
        sum += temp_number % 10
        temp_number //= 100
        lenght += 1

    # Checks if last digit of the sum is 0 and the length of the card number is correct
    if sum % 10 == 0 and lenght >= 13 and lenght <= 16:
        temp_number = card

        # Loop to get the first and second digits
        while temp_number >= 100:
            temp_number //= 10
        second_digit = temp_number % 10
        first_digit = temp_number // 10

        # Checks card manufacter based on the first two digits
        if first_digit == 4 and lenght >= 13 and lenght <= 16:
            print("VISA")
        elif first_digit == 3 and lenght == 15 and (second_digit == 4 or second_digit == 7):
            print("AMEX")
        elif first_digit == 5 and lenght == 16 and (second_digit in [1, 2, 3, 4, 5]):
            print("MASTERCARD")
        else:
            print("INVALID")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()

File: credit/credit.c
// checks the validity of a credit card number

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // gets credit card number from user
    const long NUMBER = get_long("Number: ");

    int sum = 0;
    int fdigit;
    int sdigit;
    int lenght = 0;

    // Multiply every other digit by 2 then sums the results
    long copynumber = NUMBER / 10;
    int product;
    while (copynumber > 0)
    {
        product = (copynumber % 10) * 2;
        while (product > 0)
        {
            sum += product % 10;
            product /= 10;
        }
        copynumber /= 100;
        lenght++;

    }

    // Add the previous sum to the sum of the digits that weren’t multiplied by 2
    copynumber = NUMBER;
    while (copynumber > 0)
    {
        sum += copynumber % 10;
        copynumber /= 100;
        lenght++;
    }

    // If the total’s last digit is 0 the number is valid
    if (sum % 10 == 0 && lenght >= 13 && lenght <= 16)
        // gets the first and second digit of credit card number to check the card's maker
        // also checks lenght of number
    {
        copynumber = NUMBER;
        while (copynumber >= 100)
        {
            copynumber /= 10;
        }
        sdigit = copynumber % 10;
        fdigit = copynumber / 10;

        if (fdigit == 4 && lenght >= 13 && lenght <= 16)
        {
            printf("VISA\n");
        }
        else if (fdigit == 3 && lenght == 15 && (sdigit == 4 || sdigit == 7))
        {
            printf("AMEX\n");
        }
        else if (fdigit == 5 && lenght == 16 && (sdigit == 1 || sdigit == 2 || sdigit == 3 || sdigit == 4 || sdigit == 5))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}


